### PR TITLE
Added support for HTTPS and mutual TLS (mTLS)

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message, or invoke with `--help-creds`.

This PR currently uses the prometheus-client package from its master branch.

### CODE REVIEW

**Comment by Charles1000Chen** [comment]
@andy-maier In our use case, we wanna configure the mTLS related parameters in zhmc exporter's credential yaml. Could you implement loading the mTLS parameters from credential yaml like below:
```
tls_server_config:
    ca_cert: ca.crt
    server_cert: server.crt
    server_key: server.key
```
This way/configuration is our other exporters are using. Thanks.

**Comment by andy-maier** [comment]
@Charles1000Chen I agree this is the better solution.

File: examples/hmccreds.yaml [hunk 1]
```diff
@@ -6,6 +6,14 @@ metrics:
   password: password
   verify_cert: true
 
+prometheus:
+  port: 9291
+  scheme: http
+  # server_cert_file: server_cert.pem
```
Comment (EdGue42) [inline examples/hmccreds.yaml:12]: include a comment why those lines are commented out?

File: docs/usage.rst [hunk 2]
```diff
@@ -253,6 +253,36 @@ For more information, see the
...
+for collecting metrics using HTTP GET.
+
+The parameters for the communication with Prometheus are defined in the
+HMC credentials file in the ``prometheus`` section, as in the following example:
+
+.. code-block:: yaml
+
+    prometheus:  # optional
+      port: 9291
+      scheme: https
```
Comment (Charles1000Chen) [inline docs/usage.rst:269]: @andy-maier Just want to simplify the mTLS related parameters. Could you think about omitting the "scheme"? Instead, by default use HTTP and enable HTTPS if "server_cert_file" and "server_key_file" settings are provided.

File: docs/usage.rst [hunk 3]
```diff
@@ -253,6 +253,36 @@ For more information, see the
...
+      server_cert_file: server_cert.pem
+      server_key_file: server_key.pem
+      ca_cert_file: ca_certs.pem
+      verify_cert: true
```
Comment (Charles1000Chen) [inline docs/usage.rst:273]: @andy-maier Also, hwo about omitting the "verify_cert" setting as well. Instead, use the logic that enable mTLS to verify client certificate if the "ca_cert_file" is provided together with the "server_cert_file" and "server_key_file" settings. Or, just enable simple TLS not to verify client certificate.

File: examples/hmccreds.yaml (hunk 1 above)
Comment (andy-maier) [reply examples/hmccreds.yaml:13]: Done

File: docs/usage.rst (hunk 2 above)
Comment (andy-maier) [reply docs/usage.rst]: DONE. The update removes "scheme" and implies HTTPS from the presence of "server_cert_file".

File: docs/usage.rst (hunk 3 above)
Comment (andy-maier) [reply docs/usage.rst]: DONE. The update removes "verify_cert" and implies it from the presence of "ca_cert_file". This means we can no longer use the default certificates, but they were badly documented anyway.

**Review by Charles1000Chen — APPROVED** [review]
LGTM.

**Comment by andy-maier** [comment]
I updated the PR to use the prometheus-client package from its master branch, now that PR https://github.com/prometheus/client_python/pull/946 has been merged.

**Pull request merged**